dlt/common/destination/reference.py

```diff
@@ -265,7 +296,28 @@ def _verify_schema(self) -> None:
                     logger.warning(f"A column {column_name} in table {table_name} in schema {self.schema.name} is incomplete. It was not bound to the data during normalizations stage and its data type is unknown. Did you add this column manually in code ie. as a merge key?")
 
 
-class WithStagingDataset:
+class WithStateSync(ABC):
+
+    @abstractmethod
+    def get_stored_schema(self) -> Optional[StorageSchemaInfo]:
+        """Retrieves newest schema from destination storage"""
+        pass
+
+    @abstractmethod
+    def get_stored_schema_by_hash(self, version_hash: str) -> StorageSchemaInfo:
+        pass
+
+    @abstractmethod
+    def get_stored_state(self, state_table: str, pipeline_name: str) -> Optional[StateInfo]:
+        """Loads compressed state from destination storage"""
+        pass
+
+    @abstractmethod
+    def get_stored_states(self, state_table: str) -> List[StateInfo]:
```

Comment:
we do not need that really. we just need `get_stored_state` above. it can be removed

Reply:
def test_restore_state_parallel_changes(destination_config: DestinationTestConfiguration) -> None: